bin/lang-js/src/debug.ts

```diff
@@ -1,13 +1,46 @@
-export type Debugger = (msg: any) => void;
+import process from "node:process";
+
+export type Debugger = (msg: unknown) => void;
 
 export function Debug(namespace: string): Debugger {
   const debugActive = process.env.SI_LANG_JS_LOG || process.env.SI_LOG;
-
-  return (msg: any) => {
+  return (msg: unknown) => {
     if (debugActive) {
-      const pretty_json = JSON.stringify(msg, null, 2);
-      for (const line of pretty_json.split("\n")) {
-        process.stderr.write(`${namespace} ${line}\n`);
+      try {
+        const safeStringify = (obj: unknown): string => {
```

Comment:
:heart_hands::skin-tone-3: that WeakSet, it looks like the safeStringify method could be a private fn all by itself there, rather than re-created with every Debug() call?